semgrep-core/tests/Test.ml

```diff
@@ -136,6 +138,53 @@ let regression_tests_for_lang ~with_caching files lang =
  )
 (*e: function [[Test.regression_tests_for_lang]] *)
 
+let tainting_test lang rules_file file =
+  let rules =
+    try
+      Parse_tainting_rules.parse rules_file
+    with exn ->
+      failwith (spf "fail to parse tainting rules %s (exn = %s)"
+                rules_file
+                (Common.exn_to_s exn))
+  in
+  let ast = 
+      try 
+        let { Parse_target. ast; errors; _ } = 
+          Parse_target.parse_and_resolve_name_use_pfff_or_treesitter lang file 
+        in
+        if errors <> []
+        then pr2 (spf "WARNING: fail to fully parse %s" file);
+        ast
+      with exn ->
+        failwith (spf "fail to parse %s (exn = %s)" file 
+                  (Common.exn_to_s exn))
+  in
+  let rules =
+    rules |> List.filter (fun r -> List.mem lang r.T.languages) in
+  let matches = Tainting_generic.check rules file ast in
+  let actual =
+    matches |> List.map (fun m ->
+      { E.typ = SemgrepMatchFound(m.P.rule_id.id,m.P.rule_id.message);
+        loc   = fst m.range_loc;
+        sev   = Error; }
+      )
+  in
+  let expected = Error_code.expected_error_lines_of_files [file] in
+  Error_code.compare_actual_to_expected actual expected
+
+let tainting_tests_for_lang files lang =
+  files |> List.map (fun file ->
+   (Filename.basename file) >:: (fun () ->
+    let rules_file =
+      let (d,b,_e) = Common2.dbe_of_filename file in
+      let candidate1 = Common2.filename_of_dbe (d,b,"rules") in
```

Comment:
I think you can use a .yaml instead of .rules

Reply:
I agree .yaml it's better, fixed!